// ======================================================================================
// File         : ResourceWidget.h
// Author       : Wu Jie 
// Last Change  : 04/08/2010 | 11:44:08 AM | Thursday,April
// Description  : 
// ======================================================================================

// #################################################################################
#ifndef RESOURCEWIDGET_H_1270364559
#define RESOURCEWIDGET_H_1270364559
// #################################################################################

///////////////////////////////////////////////////////////////////////////////
// includes
///////////////////////////////////////////////////////////////////////////////

#include "ui_ResourceWidget.h"

///////////////////////////////////////////////////////////////////////////////
// defines
///////////////////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////////////////
// class ResourceWidget 
// 
// Purpose: 
// 
///////////////////////////////////////////////////////////////////////////////

class ResourceWidget : public QWidget, private Ui::ResourceWidget
{
    Q_OBJECT

    ///////////////////////////////////////////////////////////////////////////////
    // public mebmer functions
    ///////////////////////////////////////////////////////////////////////////////

public:

    ResourceWidget ( QWidget* _pParent = NULL );
    ~ResourceWidget ();

    //
    bool init ();
    void deinit ();
    void explore ( const QString& _path );

    ///////////////////////////////////////////////////////////////////////////////
    // public slots 
    ///////////////////////////////////////////////////////////////////////////////

public slots:

    void refresh();

    ///////////////////////////////////////////////////////////////////////////////
    // protected slots
    ///////////////////////////////////////////////////////////////////////////////

protected slots:

    void applyFilter ( const QString& _pattern );

    ///////////////////////////////////////////////////////////////////////////////
    // protected member functions
    ///////////////////////////////////////////////////////////////////////////////

protected:

    // TODO: void doFilter ( class CResourceItem* _pItem, const QString& _pattern );

}; // end class ResourceWidget

// #################################################################################
#endif // END RESOURCEWIDGET_H_1270364559
// #################################################################################


